### PR TITLE
Rename `solution_set` to `loss_vals` in hypervolume computation

### PR DESCRIPTION
**Pull request by nabenabe0928**
## Motivation

As `loss_vals` is used frequently in the Optuna codebase and `loss_vals` makes it clear for us to recognize the better directions for each objective, I renamed `solution_set` in hypervolume computation to `loss_vals`.

## Description of the changes

- Replace `solution_set` to `loss_vals`
- Add some docs

### CODE REVIEW

**Comment by HideakiImamura** [comment]
@not522 @eukaryo Could you review this PR?

**Review by eukaryo — APPROVED** [review]
LGTM!

File: optuna/_hypervolume/wfg.py [hunk 1]
```diff
@@ -56,24 +57,37 @@ def compute_hypervolume(
...
+            An array of loss value vectors to calculate the hypervolume.
+        reference_point:
+            The reference point used to calculate the hypervolume.
+        assume_pareto:
+            Whether to assume the Pareto optimality to ``loss_vals``.
+            In other words, if ``True``, none of loss vectors are dominated by another.
+
+    Returns:
+        hypervolume:
+            The hypervolume of the given arguments.
```
Comment (not522) [inline optuna/_hypervolume/wfg.py:71]: I think this nesting is not necessary.

```suggestion
        The hypervolume of the given arguments.
```

**Review by not522 — APPROVED** [review]
LGTM!

**Pull request merged**